content/departments/engineering/dev/process/principles-and-practices.md

```diff
@@ -29,3 +29,46 @@ If you have an idea for something you want to work on, then you have a few optio
+- It's up to _you_ to come up with ideas on how to improve what you're building.
+- It's up to _you_ to think about whether what you build actually solves the customer's problem and provides value. You, as an engineer, can see possibilities and options that others can't – it's up to you to make use of that knowledge.
+- It's up to _you_ to collaborate with product, design, and analytics to ensure there is a business case for your work, it meets our user experience standards, and that analytics exist to help gauge its value to the customer.
+- It's up to _you_ to define a roadmap if one is missing and to then execute on it, to update it, and to share what you learned with others, so they can understand what you're working on and why and how it relates to their work.
+- It's up to _you_ to propose decisions that you think should be made. Instead of waiting and asking for others to make decisions for you and being blocked. And if you don't get the feedback on your proposed decisions that you need, it's up to you to let others know that you're going to execute on what you decided and then execute.
+- It's up to _you_ to take a step back when you realize that you're just building something because it's fun to build, or because it's new & shiny, but not because it has value to customers.
+- It's up to _you_ to do everything you can as an engineer to provide as much value as possible to our customers.
+
+Sometimes that's writing docs, sometimes it's getting deep into the weeds and doing performance optimizations for weeks, sometimes you should jump on a call with a customer to find out exactly what they want or what's bugging them, sometimes you need to record a demo video to make sure what you built actually lands with customers.
+
+Sometimes it's about civil disobedience. That doesn't mean you should break the law or behave against our values. No, on the contrary. It means that sometimes a process or a convention has outlived its usefulness (or goes against our values) and the only way to demonstrate that is by not doing it, and calling it out even if no one asked you. Other times it means breaking the chain of command, if, for example, you think something's going to happen that will affect a customer in a negative way but you don't think anyone's listening.
+
+Sometimes it means helping out other teams, because they're completely under water, but if they don't manage to pull this off, then your customers are impacted.
+
+Sometimes it means changing code that you don't officially own on paper – because you need the code to be changed in order to fix your customer's problem. You need team A to build FooWidget but team A is busy with something else and said they can't do it at the moment? Forget about the which-teams-owns-which-folder level of ownership and start asking: can _you_ build X? Can you build a prototype and get sign-off from team A and then continue, because your customer is waiting for it?
```

Comment:
What do you think about pointing out that you can expect teams to be open and supportive of contributions you need to make to reach your goals? Looking at the ownership from the other side's perspective — with whom you need to collaborate to get things done — your teammates and a team who owns this module that you need to change. It's somewhat obvious, but this encouragement and explicit expectation can sometimes be crucial when you meet pushback or unresponsiveness.